src/k8s-engine/images/imageProcessor.ts

```diff
@@ -504,6 +504,81 @@ export abstract class ImageProcessor extends EventEmitter {
 
     console.log(`Installing kim: kim ${ args.join(' ') }`);
 
+    // Need two loops here. First loop is to verify that we an successfully run `kim builder install`.
+    // If it fails because kubernetes is still running a docker node, we wait and retry, assuming that
+    // eventually it will stop using the docker node and will be running the newer containerd-based node.
+    // You can run `watch kubectl get node -o 'jsonpath={.items[].status.nodeInfo.containerRuntimeVersion}'`
+    // to see how this changes.
+    //
+    // If `kim builder install` fails for any other reason,
+    // this is treated as a fatal but non-terminating error -- the user is advised that they might
+    // need to reset kubernetes in order to build images with nerdctl.
+    //
+    // Once it can successfully run `docker builder install`, it moves to the next loop.
+    // Otherwise it fails on a timeout, again with a logged suggestion to reset kubernetes.
+    while (true) {
+      try {
+        await childProcess.spawnFile(
+          resources.executable('kim'),
+          args,
+          {
+            stdio:       ['ignore', 'pipe', 'pipe'],
+            windowsHide: true,
+          });
+        break;
+      } catch (e) {
+        console.error(`Failed to restart the kim builder: ${ e.message }.`);
+        if (!e.stderr?.includes('Error: container runtime `docker` not supported')) {
```

Comment:
Should we not sanity-check `backend.currentEngine`?

Reply:
I've resisted moving `installKimBuilder` into `nerdctlImageProcessor` and `uninstallKimBuilder` into `mobyImageProcessor` because I want to make it as painless as possible to support future container engines. But right now `install` is only called from `nerdctlImageProcessor` so I don't think there's a need for it.